Sources/PersistentCollections/Node/_Node+Mutations.swift

```diff
@@ -12,13 +12,19 @@
 // MARK: Node-level mutation operations
 
 extension _Node.UnsafeHandle {
-  /// Insert `item` at `slot` corresponding to `bucket`.
+  /// Make room for a new item at `slot` corresponding to `bucket`.
   /// There must be enough free space in the node to fit the new item.
   ///
   /// `itemMap` must not already reflect the insertion at the time this
   /// function is called. This method does not update `itemMap`.
+  ///
+  /// - Returns: an unsafe mutable pointer to uninitialized memory that is
+  ///    ready to store the new item. It is the caller's responsibility to
+  ///    initialize this memory.
   @inlinable
-  internal func _insertItem(_ item: __owned Element, at slot: _Slot) {
+  internal func _makeRoomForNewItem(
```

Comment:
👍

Reply:
This was such a pain to plumb through. Ideally we wouldn't need to split the action of restructuring the tree to make room for an item out from the final act of initializing storage, but it's difficult to implement `_modify` accessors well without doing something like this. 

Very soon after this function I switched to delegating initialization/deinitialization of inserted/removed items to closures that get passed in as arguments, which seems to be less painful overall. (It's still quite annoying though.)